enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,457 @@
+each availability zone.
+
+it consist in selecting only those subnets which: 1) belong to the VPC of the cluster, 2) belong to the cluster (have kubernetes.io/cluster/{cluster-name} tag), 3) have public or internal ELB role (tagged with kubernetes.io/role/elb or kubernetes.io/role/internal-elb)
+
+## Motivation
+
+Cluster admins on AWS may have dedicated subnets for their load balancers due to
+security reasons or infrastructure constraints. Currently, cluster admins can manually
+add the `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation to the service.
+However, this approach is considered bad practice since the service is managed
+by the Ingress Operator. Additionally, the annotation must be added after the
+load balancer-type service has already been created by the Ingress Operator. This
+means that an Ingress Controller may initially be deployed in a broken state.
+Cluster admins need an API for the Ingress Controller which will populate this
+annotation simultaneously with the creation of LoadBalancer-type Services.
+
+### User Stories
+
+#### Day 2 Load Balancer Subnet Selection on AWS
+
+_"As a cluster admin, when configuring an IngressController in AWS, I want to
+specify the subnet selection for the LoadBalancer-type Service on Day 2."_
+
+The cluster admin can modify the IngressController to specify a list of subnets,
+one for each availability zone, for the LoadBalancer-type Service. This is available as a
+Day-2 operation to the cluster admin.
+
+#### Load Balancer Subnet Selection Migration on AWS
+
+_"As a cluster admin, I want to migrate an IngressController load balancer-type
+service, currently using an unmanaged service.beta.kubernetes.io/aws-load-balancer-subnets
+annotation, to using the new IngressController API field so that the annotation is now
+managed."_
+
+A cluster admin manually set the `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation,
+and now wishes to OpenShift v4.17 to use new IngressController API field, `spec.TBD`.
+
+1. Cluster admin confirms that initially `service.beta.kubernetes.io/aws-load-balancer-subnets` is set on the load
+   balancer-type service managed by the IngressController.
+2. The cluster admin upgrades OpenShift to v4.17 and the annotation is not removed or changed.
+3. The cluster admin sets `spec.TBD` on the IngressController to the same value as their annotation,
+   and the annotation is unchanged, but now is managed by the Ingress Controller.
+4. From now on, the cluster admin uses `spec.TBD` to adjust the ingress controller subnets.
+
+### Goals
+
+- Introduce a new field in the IngressController API for subnet selection in AWS for Day 2 operations.
+- Support migration from an unmanaged `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation to using the
+  new API field.
+
+### Non-Goals
+
+- Support an install-time (Day 1) option for subnet selection of the default ingress controller.
```

Comment:
Need to check with Service Delivery to verify that this should be a non-goal.

Reply:
Yes thanks - will do. I put it as an open question too.

Reply:
I posted the question on the RFE and the Epic, set the "Service Delivery Architecture Overview" and "PM Sync" Check box on the Epic, and I'll bring up the question on our PM Monday call.

But what's the right mechanism to get an answer from SD?

Reply:
Nvm, I got an answer. Service Delivery needs it as an install time option.

Reply:
For future reference, the usual way to request review from SD is to edit the epic, go to the "Escalation" tab, look under the "Discussion Needed" group, and check the "Service Delivery Architecture Overview" checkbox.